src/EnergyPlus/SystemAvailabilityManager.cc

```diff
@@ -641,10 +641,20 @@ namespace SystemAvailabilityManager {
                                                    cCurrentModuleObject,
                                                    cAlphaArgs(1),
                                                    cAlphaFieldNames(6),
-                                                   cAlphaArgs(6)));
+                                                   cAlphaArgs(4)));
                             ErrorsFound = true;
                         }
                     }
+                } else if (nightCycleMgr.nightCycleControlType == NightCycleControlType::OnControlZone) {
+                    ShowSevereError(state,
+                                    format("{}{} = \"{}\" {} required when \"{}\" = {}.",
+                                           RoutineName,
+                                           cCurrentModuleObject,
+                                           cAlphaArgs(1),
+                                           cAlphaFieldNames(6),
+                                           cAlphaFieldNames(4),
+                                           cAlphaArgs(4)));
+                    ErrorsFound = true;
```

Comment:
The program would crash if control zone was blank and control type = CycleOnControlZone. Program exits gracefully now.